protocols/client/client_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// gRPC client wrapper UT
+
+package client
+
+import (
+	"net"
+	"os"
+	"testing"
+
+	"google.golang.org/grpc"
+
+	"github.com/kata-project/agent/protocols/mockserver"
+)
+
+const mockSockAddr = "/tmp/agentserver.sock"
+const unixMockAddr = "unix://" + mockSockAddr
+const badMockAddr = "vsock://" + mockSockAddr
+
+func startMockServer(t *testing.T) (*grpc.Server, chan error, error) {
+	os.Remove(mockSockAddr)
+
+	l, err := net.Listen("unix", mockSockAddr)
+	if err != nil {
+		t.Fatalf("Listen on %s failed: %s", mockSockAddr, err)
+	}
+
+	mock := mockserver.NewMockServer()
+
+	stopWait := make(chan error, 1)
+	go func() {
+		stopWait <- mock.Serve(l)
+	}()
+
+	return mock, stopWait, nil
+}
+
+func TestNewAgentClient(t *testing.T) {
+	mock, waitCh, err := startMockServer(t)
+	if err != nil {
```

Comment:
Just an fyi - we're now using https://godoc.org/github.com/stretchr/testify/assert in Clear Containers for all new test code as simplifies the code (avoiding the overly-verbose `if err != nil { ... }` blocks everywhere :-):

```
// used to avoid having to pass the `testing.T` to the assert call every time for even more brevity.
assert := assert.New(t)
    :
    :
mock, waitCh, err := startMockServer(t)
assert.NoError(err)
```